superset/commands/dataset/update.py

```diff
@@ -86,15 +86,21 @@ def validate(self) -> None:
         except SupersetSecurityException as ex:
             raise DatasetForbiddenError() from ex
 
-        database_id = self._properties.get("database", None)
-        table_name = self._properties.get("table_name", None)
+        database_id = self._properties.get("database")
+
+        table = Table(
+            self._properties.get("table_name"),  # type: ignore
+            self._properties.get("schema"),
+            self._properties.get("catalog"),
```

Comment:
This doesn't exist in the schema, though is consistent with other logic. See https://github.com/apache/superset/commit/6cf681df6808c9b612cff1e53ddb6925a9b28ebf#r141673924 for details.